src/test/java/com/nextcloud/android/sso/api/TestRetrofitAPI.java

```diff
@@ -64,6 +64,16 @@ public void setUp() {
         mApi = new NextcloudRetrofitApiBuilder(nextcloudApiMock, mApiEndpoint).create(API.class);
     }
 
+    @Test
+    public void callWithNoReturnType() throws Exception {
+        mApi.callWithNoReturnType();
+        NextcloudRequest request = new NextcloudRequest.Builder()
+                .setMethod("GET")
+                .setUrl(mApiEndpoint + "callWithNoReturnType")
+                .build();
+        verify(nextcloudApiMock).performRequestV2(eq(retrofit2.Call.class), eq(request));
```

Comment:
Doesn't this just verify the call is made? The problem is returning the async result, so I'm not sure if this a viable test... Does this test break if you bring the `map()` back?

Reply:
Well.. If you take a look at the map operator, it has a null-check inside (`Object.ensureNotNull()`). Therefore we can't use it 🤔

Reply:
Yes, that is indeed a problem. The data is coming from a mock and therefore the observable always returns an empty observable. If we want to add more checks I think we'd have to add dummy objects for all types.. right?

Reply:
I think a mocked test isn't the way to go here, at least we are mocking the wrong objects here... The mock would need to be much deeper, somewhere at the NC Files app interface I guess. We need a mock which instantly triggers the callbacks to see, if anything dies in this chain, but I didn't dig in this far yet. I'll have a look!

Reply:
I agree, the tests here are mostly to check that the type and url was extracted correctly and that the interface was called. However return types are not considered. Thank you for looking into it!